components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java

```diff
@@ -245,15 +247,15 @@ public void unregisterHookSupplier(final Supplier<StorageTxHook> hookSupplier) {
           tx.commit();
           return result;
         }
-        catch (OConcurrentModificationException e) {
-          lastConcurrentException = e;
+        catch (OConcurrentModificationException | ORecordDuplicatedException e) {
```

Comment:
The places which currently use the `perform` method are all create-or-update - if we don't retry on this exception then you can get failures when two threads attempt to create the same logical entity at exactly the same time.

When we switch to the method aspect solution we'll handle create-only cases differently.